krypto/src/commonMain/kotlin/com/soywiz/krypto/encoding/Base64.kt

```diff
@@ -77,15 +102,51 @@ object Base64 {
             out.append('=')
         }
 
-        return out.toString()
+        return if (url) {
+            base64ToBase64Url(out.toString(), doPadding)
+        } else {
+            out.toString()
+        }
+    }
+
+    /**
+     * Converts a Base64Url encoded ByteArray to a Base64 encoded ByteArray without adding padding.
+     */
+    private fun base64UrlToBase64(src: ByteArray) {
+        src.forEachIndexed { index, byte ->
+            if (byte == '-'.code.toByte()) {
+                src[index] = '+'.code.toByte()
```

Comment:
We shouldn't mutate the source array, or that could cause unexpected behaviors. For performance we shouldn't either make a copy, so we should figure out a performant way to handle this while decoding

Reply:
I see your point. The prior approach was making a copy. One option is to create a 2nd version of TABLE and DECODE that uses the URL character set, and then use it during encoding/decoding. That would have similar speed performance, with the tradeoff that now the Base64 object has 128 bytes of extra tables. Unless someone is targeting a very low storage/memory embedded system I don't envision that being a problem.

Reply:
I decided it would be worth the extra 64 bytes to also not create a copy of the encoded data. So implemented a separate TABLE for URL encoding as well.